test/hotspot/jtreg/runtime/ErrorHandling/HsErrFileUtils.java

```diff
@@ -60,46 +61,86 @@ public static File openHsErrFileFromOutput(OutputAnalyzer output) {
     }
 
     /**
-     * Given an open hs-err file, read it line by line and check for pattern. Pattern
-     * need to appear in order, but not necessarily uninterrupted.
+     * Given an open hs-err file, read it line by line and check for existence of a pattern. Will fail
+     * if pattern are missing, or if the END marker is missing.
+     * @param f input file
+     * @param patterns an array of patterns that need to match, in that order
+     * @param verbose if true, the content of the hs-err file is printed while matching. If false, only the matched patterns
+     *                are printed.
+     * @throws RuntimeException, {@link IOException}
      */
     public static void checkHsErrFileContent(File f, Pattern[] patterns, boolean verbose) throws IOException {
+        checkHsErrFileContent(f, patterns, null, true, verbose);
+    }
 
-        FileInputStream fis = new FileInputStream(f);
-        BufferedReader br = new BufferedReader(new InputStreamReader(fis));
-        String line = null;
-
-        int currentPattern = 0;
-
-        String lastLine = null;
-        while ((line = br.readLine()) != null) {
-            if (verbose) {
-                System.out.println(line);
+    /**
+     * Given an open hs-err file, read it line by line and check for various conditions.
+     * @param f input file
+     * @param positivePatterns Optional array of patterns that need to appear, in given order, in the file. Missing
```

Comment:
Consistency check: Sometimes you start the description with a capital letter and other times not. I don't know what preferred JDK Java style is.